### ISSUE TITLE
Support :ensure-system-package

### ISSUE THREAD

**Issue by fernandodemorais-jf** [conao3/leaf-keywords.el#138]
Hello,

Any possibility to add support for the `:ensure-system-package` keyword in `leaf`, using [system-package](https://gitlab.com/jabranham/system-packages/tree/05add2fe051846e2ecb3c23ef22c41ecc59a1f36) as a dependency, in a similar way that `use-package` do?

It's the only reason that holds me in `use-package`.

Thanks!

**Comment by conao3** [comment]
> It's the only reason that holds me in use-package.

This is a good reason to tackle this :)
I implement `:ensure-system-package`, please try it (after MELPA update (maybe 2 or 3 hours))

**Comment by fernandodemorais-jf** [comment]
Thank you!

It worked like a charm. I'll prepare my complete migration to `leaf`.

Cheers!

**Comment by fernandodemorais-jf** [comment]
Hello,

Reopening the issue just to know if its possible to add two missing features of `:ensure-system-package` that `use-package` contains, which is something like the following:

```elisp
;; Specify the package the package name for given symbol
((leaf leaf
       :ensure-system-package (rg . ripgrep)
       :config (leaf-init))
     (prog1 'leaf
       (unless (executable-find "rg") (system-packages-install "ripgrep"))
       (leaf-init)))

;; Install package if the presence of file is nil
((leaf vterm
       :ensure t
       :ensure-system-package (("/usr/lib/libvterm.so" . libvterm)
                               (cmake libtool)))
     (prog 1 'vterm
       (unless (file-exists-p "/usr/lib/libvterm.so") (system-packages-install "libvterm))
       (unless (executable-find "cmake") (system-packages-install "cmake"))
       (unless (executable-find "libtool") (system-packages-install "libtool"))))
```

I hope that implementation is possible.

**Comment by conao3** [comment]
Thanks for mentioning the missing features and these testcases! I implemented at #141, please try it.

**Comment by fernandodemorais-jf** [comment]
Thank you once more.

Its all working.

Cheers!
